include/aspect/boundary_composition/interface.h

```diff
@@ -215,15 +219,26 @@ namespace aspect
         get_active_boundary_composition_conditions () const;
 
         /**
-         * Go through the list of all boundary composition models that have been selected in
-         * the input file (and are consequently currently active) and see if one
-         * of them has the desired type specified by the template argument. If so,
-         * return a pointer to it. If no boundary composition model is active
-         * that matches the given type, return a NULL pointer.
+         * Go through the list of all boundary composition models that have been selected
+         * in the input file (and are consequently currently active) and return
+         * true if one of them has the desired type specified by the template
+         * argument.
+         */
+        template <typename BoundaryCompositionType>
+        bool
+        has_matching_boundary_composition_model () const;
+
+        /**
+         * Go through the list of all boundary composition models that have been selected
+         * in the input file (and are consequently currently active) and see
+         * if one of them has the type specified by the template
+         * argument or can be casted to that type. If so, return a reference
+         * to it. If no boundary composition model is active that matches the given type,
+         * throw an exception.
          */
         template <typename BoundaryCompositionType>
-        BoundaryCompositionType *
-        find_boundary_composition_model () const;
```

Comment:
You are removing this function. Has this not been used/tested before?

Reply:
It was seemingly never used before, but I added it back in with a deprecation warning